src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            position_matrix = translation_matrix(self.position)
+            self.mesh.apply_transform(position_matrix)
+
+        self.orientation = orientation
+
+    @classmethod
+    def fromFile(cls, path, filetype=None, compressed=None, **kwargs):
+        """Load a region from a file, attempting to infer filetype and compression.
+
+        For example: "foo.obj.bz2" is assumed to be a compressed .obj file.
+        "foo.obj" is assumed to be an uncompressed .obj file. "foo" is an
+        unknown filetype, so unless a filetype is provided an exception will be raised.
+
+        Args:
+            path: Path to the file to import
+            filetype: Filetype of file to be imported. This will be inferred if not provided
+            compressed: Wheter or not this file is compressed (with bz2). This will be inferred
+                if not provided
+            kwargs: Additional arguments to the MeshRegion initializer.
+        """
+        mesh = loadMesh(path, filetype, compressed)
+        return cls(mesh=mesh, **kwargs)
+
+    ## API Methods ##
+    # Mesh Access #
+    @property
+    def mesh(self):
+        # Prevent access to mesh unless it actually represents region.
+        if isLazy(self):
+            raise ValueError("Attempting to access the Mesh of an unsampled MeshRegion.")
+
+        return self._mesh
+
+    def findOn(self, point, on_direction):
+        """ Find the nearest point in the region following the on_direction.
+
+        Returns None if no such points exist.
+        """
+        assert not needsSampling(self)
+
+        # Check if this region contains the point, in which case we simply
+        # return the point.
+        if self.containsPoint(point):
+            return point
+
+        # Get first point hit in both directions of ray
+        point = point.coordinates
+
+        if on_direction is not None:
+            on_direction = numpy.array(on_direction)
+        else:
+            if isinstance(self, MeshVolumeRegion):
+                on_direction = numpy.array((0,0,1))
+            elif isinstance(self, MeshSurfaceRegion):
+                on_direction = sum(self.mesh.face_normals*self.mesh.area_faces[:, numpy.newaxis])/self.mesh.area
+                on_direction = on_direction/numpy.linalg.norm(on_direction)
+
+        intersection_data, _, _ = self.mesh.ray.intersects_location(ray_origins=[point, point], \
+            ray_directions=[on_direction, -1*on_direction], multiple_hits=False)
+
+        if len(intersection_data) == 0:
+            return None
+
+        # Get point with least euclidean distance
+        def euclidean_distance(p_1, p_2):
+            diff_list = [p_1[i] - p_2[i] for i in range(3)]
+            square_list = [math.pow(p, 2) for p in diff_list]
+            return math.sqrt(sum(square_list))
+
+        distances = [euclidean_distance(point, p) for p in intersection_data]
+
+        closest_point = intersection_data[distances.index(min(distances))]
```

Comment:
After switching to numpy above:
```suggestion
        closest_point = intersection_data[numpy.argmin(distances)]
```